src/persist/src/nemesis/direct.rs

```diff
@@ -289,9 +296,11 @@ impl Direct {
     fn stop(&mut self) -> Result<(), Error> {
         let res = self.persister.stop();
 
-        // Stopping the persister should allow the dataflows to finish.
-        while self.worker.0.step() {}
-
+        // Stopping the persister allows the dataflows to finish.
+        for (_, dataflow) in self.dataflows.drain() {
+            dataflow.progress_handle.close();
+            dataflow.workers.0.join();
```

Comment:
super minor: this tripped me up a tiny bit because I misread it as "only join on worker 0" -- might be worth a clarifying comment?

Reply:
added a join method to TimelyWorkers that proxies to the real join instead